From 3cb325a5954cbdf33b6741f19386a4880a7557ee Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:38:39 -0400
Subject: [alsa] hda: add capture mute led support in led_power_filter

Message-id: <140972823573-385-patchset-jkysela@redhat.com>
Patchwork-id: 91081
O-Subject: [RHEL 7.1 PATCH 384/442] ALSA: hda - add capture mute led support in led_power_filter
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Now the led_power_filter() needs to handle 3 situations:
- only mute_led_nid is set
- only cap_mute_led_nid is set
- both mute_led_ind and cap_mute_led_nid are set

BugLink: https://bugs.launchpad.net/bugs/1329580
Cc: David Henningsson <david.henningsson@canonical.com>
Cc: Kailang Yang <kailang@realtek.com>
Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 50dd9050e45ce11c575eabcaee8a914caf78fb74)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 73b1759..512582b 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -3403,7 +3403,8 @@ static unsigned int led_power_filter(struct hda_codec *codec,
 {
  struct alc_spec *spec = codec->spec;
 
- if (power_state != AC_PWRST_D3 || nid != spec->mute_led_nid)
+ if (power_state != AC_PWRST_D3 || nid == 0 ||
+     (nid != spec->mute_led_nid && nid != spec->cap_mute_led_nid))
   return power_state;
 
  /* Set pin ctl again, it might have just been set to 0 */
@@ -3563,6 +3564,7 @@ static void alc269_fixup_hp_gpio_mic1_led(struct hda_codec *codec,
   spec->gpio_led = 0;
   spec->cap_mute_led_nid = 0x18;
   snd_hda_add_verbs(codec, gpio_init);
+  codec->power_filter = led_power_filter;
  }
 }
 
-- 
1.7.1